src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,107 @@
+package duke.storage;
+
+import duke.task.Task;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Storage {
+    private final Path filePath;
+
+    public Storage() {
+        this.filePath = Paths.get(System.getProperty("user.dir"), "data.txt");
+    }
+
+    /**
+     * Returns a Task object that has been deserialized from its string representation.
+     * @param serializedTask Serialized Task object (String representation of Task object)
+     * @return Task object
+     */
+
+    public Task deserializeTask(String serializedTask) throws RuntimeException {
+        String[] tokens = serializedTask.split(" \\| ");
+
+        String taskIndex = tokens[0];
+        boolean isDone = tokens[1].equals("1");
+        String description = tokens[2];
+
+        try {
+            switch (taskIndex) {
+            case ("T"):
+                return new ToDo(description, isDone);
+            case ("D"):
+                return new Deadline(description, isDone, new SimpleDateFormat("y-M-d").parse(tokens[3]));
+            case ("E"):
+                return new Event(description, isDone, new SimpleDateFormat("y-M-d").parse(tokens[3]));
+            default:
+                throw new RuntimeException("Task is invalid. File might be corrupted.");
+            }
+        } catch (ParseException e) {
+            throw new RuntimeException("Date can't be parsed.");
+        }
+    }
```

Comment:
It's amazing that you added a custom serialization/deserialization helper to write an editable text file. This makes your text file editable as long as the format of the file is followed.

I chose to use default serialization libraries for to save the entire TaskList object inside, and upon a new write onto the TaskList, to overwrite that file. This is so that people don't tamper with the saved data file and also prevents Storage class from handling logic related to the the format of the content in the file. I didn't know how to value to your PR, so just giving a possible alternative here.
![image](https://user-images.githubusercontent.com/38996397/91786926-59151180-ec3b-11ea-8988-fe08f4654561.png)

Reply:
Thank you @rtshkmr for the information. I will definitely consider it 👍 

P.S. you have definitely added value to the PR; your comments are constructive and helpful. Thank you! 😄